### PR TITLE
Ported CreateMD algorithm to C++ from Python

### LINKED ISSUE #14147: Port the CreateMD algorithm from Python to C++
This is necessary as CreateMD will be used as a child algorithm by AccumulateMD. Unit tests for AccumulateMD are hitting the problem `Algorithm not registered CreateMD` #13507.

### PR DESCRIPTION
**Pull request by matthew-d-jones**
Fixes #14147 

The CreateMD algorithm was ported to C++.
The PR is urgent as it blocks #13507.

**For Tester:**
The CreateMD algorithm is documented [here](http://docs.mantidproject.org/nightly/algorithms/CreateMD-v1.html)
The following python can be used to test. The nexus file is in the test data.

``` python
CreateMDTest_mdworkspace = CreateMD("CNCS_7860_event.nxs", EFix='12.0', Alatt=[1,1,1], Angdeg=[90,90,90], u=[1,0,0], v=[0,1,0])
```

### CODE REVIEW

**Comment by martyngigg** [comment]
@matthew-d-jones Can you include the appropriate `Fixes #` line?

File: Framework/MDAlgorithms/src/CreateMD.cpp [hunk 1]
```diff
@@ -0,0 +1,468 @@
...
+       ++ws_iter) {
+    AnalysisDataService::Instance().remove(*ws_iter);
+  }
+
+  this->setProperty("OutputWorkspace", output_workspace);
+}
+
+/*
+ * Load data from file
+ */
```
Comment (picatess) [inline Framework/MDAlgorithms/src/CreateMD.cpp]: You can use the doxygen way of annotation with @param , @returns, etc.

File: Framework/MDAlgorithms/src/CreateMD.cpp [hunk 2]
```diff
@@ -0,0 +1,468 @@
...
+Mantid::API::Workspace_sptr CreateMD::loadWs(const std::string &filename,
+                                             const std::string &wsname) {
+  Algorithm_sptr load_alg = createChildAlgorithm("Load");
+
```
Comment (picatess) [inline Framework/MDAlgorithms/src/CreateMD.cpp]: your methods are not changing any data members, so you can declare them `const`.

File: Framework/MDAlgorithms/inc/MantidMDAlgorithms/CreateMD.h [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+#include "MantidDataObjects/MDEventWorkspace.h"
+namespace Mantid {
+namespace MDAlgorithms {
+
+void padParameterVector(std::vector<double> &param_vector,
+                        const unsigned long grow_to_size);
+
+bool any_given(const std::vector<std::vector<double>> &params);
+
+bool all_given(const std::vector<std::vector<double>> &params);
```
Comment (picatess) [inline Framework/MDAlgorithms/inc/MantidMDAlgorithms/CreateMD.h]: Nice

**Comment by picatess** [comment]
Testing worked withoug any issues. Happy to merge

**Pull request merged**